### PR TITLE
Build Script to Automate the Pytorch DINO Model on NVIDIA Container

### PR DESCRIPTION
**Pull request by raj-prince**
Purpose: This is required to test the gcsfuse read workload under the real life ML environment.

Please refer the previous comments on this PR: https://github.com/GoogleCloudPlatform/gcsfuse/pull/871.

### CODE REVIEW

File: perfmetrics/scripts/ml_tests/pytorch/dino/setup_container.sh [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+sed -i "$lines"'d' $folder_file
+sed -i "$x"'r bypassed_code.py' $folder_file
+
+# Fix the caching issue - comes when we run the model first time with 8
+# nproc_per_node - by downloading the model in single thread environment.
+python -c 'import torch;torch.hub.list("facebookresearch/xcit:main")'
+
+# Run the pytorch Dino model
+# We need to run it in foreground mode to make the container running.
+# Todo: Change mini_imagenet/train to imagenet/ILSVRC/Data/CLS-LOC/train/
```
Comment (raj-prince) [inline perfmetrics/scripts/ml_tests/pytorch/dino/setup_container.sh:60]: Change the data-set path and epoch size - just before the submission.

Comment (sethiay) [reply perfmetrics/scripts/ml_tests/pytorch/dino/setup_container.sh:60]: Also, change the --nproc_per_node=2 to --nproc_per_node=8

File: perfmetrics/scripts/ml_tests/pytorch/dino/README-usage.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+docker system, nvidia-driver (gpu based ml training), and some utilities like,
+curl, ca-certificates, lsb-release etc.
+
+### File: perfmetrics/scripts/ml_test/pytorch/dino/setup_container.sh
+This script contains the instruction to install gcsfuse, mount GCS-bucket
+using gcsfuse, and finally runs the pytorch dino model.
+
+### File: perfmetrics/scripts/continuous_test/pytorch/dino/build.sh
+This is the parent script of the above two scripts. Firstly, it sets-up the host
+machine after that it creats the docker-image and finally it runs the container
```
Comment (sethiay) [inline perfmetrics/scripts/ml_tests/pytorch/dino/README-usage.md:21]: nit: creats -> creates

File: perfmetrics/scripts/ml_tests/pytorch/dino/README-usage.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+with the inststructions written in the setup_container.sh.
+
+## Artifacts after the Executions:
+As a kokoro build execution, we create various logs - (a) GCSFuse logs (b) Dino
```
Comment (sethiay) [inline perfmetrics/scripts/ml_tests/pytorch/dino/README-usage.md:25]: nit: I think better way could be "After the execution of kokoro job, we copy two types of logs - "

File: perfmetrics/scripts/ml_tests/pytorch/dino/README-usage.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+model logs.
+
+### GCSFuse Logs: container_artifacts/gcsfuse_logs
+We mount the gcsfuse with debug flags, this folder contains the running gcsfuse
+logs. This will be beneficial for debugging purpose.
+
+### Dino Model Logs: container_artifacts/dino-experiment/
+checkpoint*.pth - Model checkpointing. 
+log.txt - Contains the standard ouput we get after execution of DINO model.
```
Comment (sethiay) [inline perfmetrics/scripts/ml_tests/pytorch/dino/README-usage.md:34]: I believe the standard output is going to sudo docker logs -f pytorch_automation_container?

File: perfmetrics/scripts/ml_tests/pytorch/dino/README-usage.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+### Steps to debug Offline
```
Comment (sethiay) [inline perfmetrics/scripts/ml_tests/pytorch/dino/README-usage.md:36]: nit: "Steps to run the model on VM" OR "Steps to debug on VM"

File: perfmetrics/scripts/continuous_test/ml_tests/pytorch/dino/continuous.cfg [hunk 6]
```diff
@@ -0,0 +1,13 @@
+build_file: "gcsfuse/perfmetrics/scripts/continuous_test/ml_tests/pytorch/dino/build.sh"
+
+# Setting the 16 days (23040 mins) timeout for kokoro-jobs.
+timeout_mins: 23040
+
+
+# Post the gcsfuse logs as an artifact to GCSBucket
+action {
+  define_artifacts {
+    regex: "github/gcsfuse/container_artifacts/**"
```
[the remark this answers is not in the document]
Comment (raj-prince) [reply perfmetrics/scripts/continuous_test/ml_tests/pytorch/dino/continuous.cfg:10]: This is already done by KOKORO itself. 
Current path: gcsfuse-ml-automation-logs/pytorch/dino/qa/gcsfuse/ml_tests/pytorch/dino/continuous/24/20230122-020106

I have added changes to skip the top two folder in kokoro.

File: perfmetrics/scripts/ml_tests/pytorch/dino/README-usage.md (hunk 4 above)
Comment (raj-prince) [reply perfmetrics/scripts/ml_tests/pytorch/dino/README-usage.md:34]: Yes, it contains the model learning parameter value of every completed epoch.

**Review by sethiay — APPROVED** [review]

**Pull request merged**